engine/source/elements/solid/solide4_sfem/s4volnod3.F

```diff
@@ -60,53 +62,92 @@ SUBROUTINE S4VOLNOD3(VOLNOD6,X,NC1,NC2,NC3,NC4,OFFG,SAV,NEL)
 C     REAL
 
       my_real
-     .   X41, Y41, Z41, X42, Y42, Z42, X43, Y43, Z43
-      my_real
-     .   B1(MVSIZ), C1(MVSIZ), D1(MVSIZ), DET(MVSIZ),
+     .   DET(MVSIZ)
+      DOUBLE PRECISION
+     .   X41, Y41, Z41, X42, Y42, Z42, X43, Y43, Z43,
+     .   B1, C1, D1
+      DOUBLE PRECISION
      .   X1(MVSIZ), X2(MVSIZ), X3(MVSIZ), X4(MVSIZ),
      .   Y1(MVSIZ), Y2(MVSIZ), Y3(MVSIZ), Y4(MVSIZ),
      .   Z1(MVSIZ), Z2(MVSIZ), Z3(MVSIZ), Z4(MVSIZ)
 
       DOUBLE PRECISION
-     .   DET6(6,MVSIZ)
+     .   DET6(6,MVSIZ),DETDP
 C----------------------------
 C     NODAL COORDINATES     
 C----------------------------
       IF (ISMSTR==1.OR.ISMSTR==11) RETURN
-      IF((ISMSTR==2.OR.ISMSTR==12).AND.IDTMIN(1)==3)THEN
-       DO I=LFT,LLT
-       IF(ABS(OFFG(I))>ONE)THEN
-       ELSE
-        X1(I)=X(1,NC1(I))
-        Y1(I)=X(2,NC1(I))
-        Z1(I)=X(3,NC1(I))
-        X2(I)=X(1,NC2(I))
-        Y2(I)=X(2,NC2(I))
-        Z2(I)=X(3,NC2(I))
-        X3(I)=X(1,NC3(I))
-        Y3(I)=X(2,NC3(I))
-        Z3(I)=X(3,NC3(I))
-        X4(I)=X(1,NC4(I))
-        Y4(I)=X(2,NC4(I))
-        Z4(I)=X(3,NC4(I))
-       ENDIF
-       ENDDO
+      IF(IRESP == 1) THEN
+        IF((ISMSTR==2.OR.ISMSTR==12).AND.IDTMIN(1)==3)THEN
+         DO I=LFT,LLT
+         IF(ABS(OFFG(I))>ONE)THEN
+         ELSE
+          X1(I)=XDP(1,NC1(I))
+          Y1(I)=XDP(2,NC1(I))
+          Z1(I)=XDP(3,NC1(I))
+          X2(I)=XDP(1,NC2(I))
+          Y2(I)=XDP(2,NC2(I))
+          Z2(I)=XDP(3,NC2(I))
+          X3(I)=XDP(1,NC3(I))
+          Y3(I)=XDP(2,NC3(I))
+          Z3(I)=XDP(3,NC3(I))
+          X4(I)=XDP(1,NC4(I))
+          Y4(I)=XDP(2,NC4(I))
+          Z4(I)=XDP(3,NC4(I))
+         ENDIF
+         ENDDO
+        ELSE
+         DO I=LFT,LLT
+          X1(I)=XDP(1,NC1(I))
+          Y1(I)=XDP(2,NC1(I))
+          Z1(I)=XDP(3,NC1(I))
+          X2(I)=XDP(1,NC2(I))
+          Y2(I)=XDP(2,NC2(I))
+          Z2(I)=XDP(3,NC2(I))
+          X3(I)=XDP(1,NC3(I))
+          Y3(I)=XDP(2,NC3(I))
+          Z3(I)=XDP(3,NC3(I))
+          X4(I)=XDP(1,NC4(I))
+          Y4(I)=XDP(2,NC4(I))
+          Z4(I)=XDP(3,NC4(I))
+         ENDDO
+        ENDIF
       ELSE
-       DO I=LFT,LLT
-        X1(I)=X(1,NC1(I))
-        Y1(I)=X(2,NC1(I))
-        Z1(I)=X(3,NC1(I))
-        X2(I)=X(1,NC2(I))
-        Y2(I)=X(2,NC2(I))
-        Z2(I)=X(3,NC2(I))
-        X3(I)=X(1,NC3(I))
-        Y3(I)=X(2,NC3(I))
-        Z3(I)=X(3,NC3(I))
-        X4(I)=X(1,NC4(I))
-        Y4(I)=X(2,NC4(I))
-        Z4(I)=X(3,NC4(I))
-       ENDDO
-      ENDIF
+        IF((ISMSTR==2.OR.ISMSTR==12).AND.IDTMIN(1)==3)THEN
+         DO I=LFT,LLT
```

Comment:
Indentation issue for the block starting here

Reply:
done